churn-risk/src/app.py

```diff
@@ -1,184 +1,193 @@
+        contents = f.read()
+
+    py_lexer = get_lexer_by_name("python")
+    html_formatter = HtmlFormatter(full=True, style="xcode")
+    q.page['code'] = ui.frame_card(box=ui.box('code', height='calc(100vh - 155px)'), title='', content=highlight(contents, py_lexer, html_formatter))
+
+def init(q: Q):
+    q.page['meta'] = ui.meta_card(box='', title='Telco Churn Analytics', layouts=[
+        ui.layout(breakpoint='xs', zones=[
+            ui.zone('header'),
+            ui.zone('title'),
+            ui.zone('content', zones=[
+                ui.zone('code'),
+                ui.zone('top', direction=ui.ZoneDirection.ROW, zones=[
+                    ui.zone('top-plot', size='70%'),
+                    ui.zone('top-stats')
+                ]),
+                ui.zone('middle', direction=ui.ZoneDirection.ROW),
+            ])
+        ])
+    ])
+    q.page['header'] = ui.header_card(
+        box='header',
+        title='Telecom Churn Analytics',
+        subtitle='EDA & Churn Modeling with AutoML & Wave',
+        nav=[
+            ui.nav_group('Main Menu', items=[
+                ui.nav_item(name='#analysis', label='Analysis'),
+                ui.nav_item(name='#code', label='Application Code'),
+            ])
+        ]
+    )
+    q.page['title'] = ui.section_card(
+        box='title',
+        title='Customer profiles from model predictions',
+        subtitle='Customer: No customer chosen',
+        items=[
+            # TODO: Replace with dropdown after https://github.com/h2oai/wave/pull/303 merged.
+            ui.picker(
+                name='customers',
+                label='Customer Phone Number',
+                choices=[ui.choice(name=str(phone), label=str(phone)) for phone in df['Phone_No']],
+                max_choices=1,
+                trigger=True
+            ),
+            ui.toggle(name='theme', label='Dark Theme', trigger=True)
+        ]
     )
 
 
-def get_figure_layout():
-    return go.Layout(margin=go.layout.Margin(l=0, r=0, b=0, t=0, pad=0, autoexpand=True), height=120)
-
-
-async def initialize(q: Q):
-    # Initialize H2O-3 model and tests data set
-    churn_predictor.build_model(config.training_data_url, config.default_model)
-    churn_predictor.set_testing_data_frame(config.testing_data_url)
-    churn_predictor.predict()
-
-    q.app.header_png = await q.site.upload([config.image_path])
-    q.app.training_file_url = await q.site.upload([config.working_data])
-    q.page['meta'] = ui.meta_card(box='', title='Telco Churn Analytics')
-    q.client.app_initialized = True
-
-
-@app("/")
+@app('/')
 async def serve(q: Q):
-    render_header(q)
-    if not q.client.app_initialized:
-        await initialize(q)
-
-    if q.args['#'] == 'tour':
-        cleanup_pages(q)
-        q.page["content"] = ui.form_card(box=config.boxes["content"], items=python_code_content("app.py"))
+    if not q.client.initialized:
+        init(q)
+        q.client.primary_color = '$blue'
```

Comment:
Could these definitions go in the `init` function for a cleaner serve function?

Reply:
sure, done